spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/sql/DefaultCloudSqlJdbcInfoProvider.java

```diff
@@ -54,12 +58,49 @@ public String getJdbcUrl() {
             this.properties.getDatabaseName(),
             this.properties.getInstanceConnectionName());
 
+    // Build additional JDBC url parameters from the configuration.
```

Comment:
Could we also add testing for these new properties in https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/f5879d92d81d5b480d2c402bd2e9e792832ebca3/spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/sql/CloudSqlEnvironmentPostProcessorTests.java#L41

Reply:
This test method should do the trick:
```java
@Test
void testCloudSqlSpringDatasourceWithAllOptions(){
    this.contextRunner
            .withPropertyValues(
                    "spring.cloud.gcp.sql.database-name=test-database",
                    "spring.cloud.gcp.sql.instance-connection-name=tubular-bells:singapore:test-instance",
                    "spring.cloud.gcp.sql.ip-types=PRIVATE",
                    "spring.cloud.gcp.sql.enable-iam-auth=true",
                    "spring.cloud.gcp.sql.target-principal=target-principal",
                    "spring.cloud.gcp.sql.delegates=delegate1,delegate2",
                    "spring.cloud.gcp.sql.admin-root-url=someAdminRootUrlValue",
                    "spring.cloud.gcp.sql.admin-service-path=someAdminServicePathValue",
                    "spring.cloud.gcp.sql.admin-quota-project=someAdminQuotaProjectValue",
                    "spring.cloud.gcp.sql.universe-domain=someUniverseDomainValue",
                    "spring.cloud.gcp.sql.refresh-strategy=lazy",
                    "spring.datasource.username=foo",
                    "spring.datasource.password=bar")
            .run(
                    context -> {
                        HikariDataSource dataSource = (HikariDataSource) context.getBean(DataSource.class);
                        assertThat(dataSource)
                                .returns("com.mysql.cj.jdbc.Driver", HikariDataSource::getDriverClassName)
                                .returns("foo", HikariDataSource::getUsername)
                                .returns("bar", HikariDataSource::getPassword)
                                .extracting(HikariDataSource::getJdbcUrl)
                                .asInstanceOf(InstanceOfAssertFactories.STRING)
                                .startsWith("jdbc:mysql://google/test-database?"
                                                    + "socketFactory=com.google.cloud.sql.mysql.SocketFactory")
                                .satisfies(jdbcUrl -> assertThat(jdbcUrl.split("&"))
                                        .containsAnyOf("cloudSqlInstance=tubular-bells:singapore:test-instance",
                                                       "ipTypes=PRIVATE",
                                                       "enableIamAuth=true",
                                                       "cloudSqlTargetPrincipal=target-principal",
                                                       "cloudSqlDelegates=delegate1%2Cdelegate2",
                                                       "cloudSqlAdminRootUrl=someAdminRootUrlValue",
                                                       "cloudSqlAdminServicePath=someAdminServicePathValue",
                                                       "cloudSqlAdminQuotaProject=someAdminQuotaProjectValue",
                                                       "cloudSqlUniverseDomain=someUniverseDomainValue",
                                                       "cloudSqlRefreshStrategy=lazy"));
                        assertThat(getSpringDatasourceDriverClassName(context))
                                .matches("com.mysql.cj.jdbc.Driver");
                    });
}
```
I tried to create a PR to the existing PR, but I have no clue about your formatting rules (spotless?)
@hessjcg , maybe you accept this "PR" by just copy the code :laughing:

Reply:
Okay, I tried with https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/3313 :sweat_smile:

Reply:
Thank you @mieseprem for the test! I have moved it code into this PR.